src/proc.h

```diff
@@ -11,7 +11,7 @@
 #include <termios.h>
 #include <unistd.h>
 
-#include <list>
+#include <deque>
```

Comment:
Have you tried vector? Any benchmarks?

From all the benchmarks I've seen, list is pretty much useless for almost all workloads (cache misses were the main issue IIRC), but vector is often nice.

See e.g. https://baptiste-wicht.com/posts/2012/12/cpp-benchmark-vector-list-deque.html and https://www.youtube.com/watch?v=YQs6IC-vgmo.

Also if both performed the same we should prefer vector because we're already using it.

Reply:
`deque` is like a `vector` but with more efficient insert at the head too, not just the end. All our inserts were to the top of the list, so we should have actually changed this to a vector *plus* changed our view of the jobs list from top-to-bottom to bottom-to-top (just using `rbegin()` and `rend()` in iteration would have sufficed), but we can do that some other time.

Reply:
Yeah, fair enough!